framework/source/class/qx/ui/form/core/VirtualDropDownList.js

```diff
@@ -393,15 +393,7 @@ qx.Class.define("qx.ui.form.core.VirtualDropDownList",
       }
       else
       {
-        var nativeArray = target.toArray();
-        qx.lang.Array.removeAll(nativeArray);
-        for (var i = 0; i < source.getLength(); i++) {
-          nativeArray.push(source.getItem(i));
-        }
-        target.length = nativeArray.length;
-
-        // necessary for keeping preselected items in sync
-        target.fireDataEvent("change", {});
+        target.splice.apply(target, [0, target.length].concat(source.toArray())).dispose();
       }
```

Comment:
Sorry, it's the first time I contribute to an open source project. :sweat_smile:
I've made a new commit with the line splitted and comments.

I use splice because this is whats I think is the closest to the previous code (remove all from target, add all from source and send an event).

I don't use it directly but with apply because if you try to do `target.splice(0, target.length, source.toArray()` you will get an array of array (`[[item1, item2, ...]]`) instead of an array containing all the items of source(`[item1, item2, ...]`).

The problem was that when you add objects that include the mixin `qx.data.marshal.MEventBubbling` to a `qx.data.Array`, the `_registerEventChaining` add a listener on it to propagate bubbling events but when you remove items manually the listeners aren't removed whereas splice does.